### PR TITLE
CLN simplify benchmark+add splitter

### PR DESCRIPTION
**Pull request by tomMoral**
- Remove `AugmentedDatset` which should probably be added as extra methods.
- Add a simple splitter to CV-based repetitions.
- Fix the solvers.

### CODE REVIEW

File: benchmark_utils/splitter.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                X_test = SliceDataset(
+                    dataset_subject, idx=0, indices=i_test
+                )
+                y_test = SliceDataset(
+                    dataset_subject, idx=1, indices=i_test
+                )
+                yield X_train, X_test, y_train, y_test
+
+
+class InterSubjectSplitter(BaseCrossValidator):
```
Comment (bruAristimunha) [inline benchmark_utils/splitter.py:70]: nice implementation

**Review by bruAristimunha — APPROVED** [review]
Looks super good and more clean @tomMoral!

**Comment by bruAristimunha** [comment]
I am looking the tests and maybe be related with some changes that we made on fakedataset on moabb side. 

I am not sure, if you need any help that I can provide, please ping me, I will be more than happy to help!
@tomMoral

**Comment by tomMoral** [comment]
@bruAristimunha if you could look into it today that would be great!! Otherwise I will try to fix it.

**Comment by tomMoral** [comment]
The first error will be fixed by benchopt/benchopt#702

**Comment by bruAristimunha** [comment]
The ShallowNet issue was solved; now, something related to Zhou Dataset, which is super curious! I can't reproduce it on my machine.

It looks like some package version issue (mne, braindecode, and moabb), but I am unsure of the source.

image

I am removing the Zhou dataset, considering it was added to replace simulated data (smaller moabb dataset)

**Comment by bruAristimunha** [comment]
Another hypothesis is that this is something related to the dataset itself, something changed in the source and this changes something with label (this can happen)

**Comment by bruAristimunha** [comment]
Hey @tomMoral!

I'm trying very hard to reproduce the error, but I'm not succeeding. Everything is working on my machine. I've already done everything I could think of, re-create env, run just the failing test, run all the tests locally, even in the log the error doesn't appear. I need help with the benchopt side. It is indicating an error that I can't see why

**Comment by bruAristimunha** [comment]
Thank you for including this logger, but I still need help seeing the error log @tomMoral.

What is most strange for me is that: 

No error when benchopt test with install:

../../../../../usr/share/miniconda3/envs/test_env/lib/python3.9/site-packages/benchopt/tests/test_benchmarks.py::test_solver[benchmark_bci-shallowfbcspnet] **PASSED** [ 96%]

And without installation:

../../../../../usr/share/miniconda3/envs/test_env/lib/python3.9/site-packages/benchopt/tests/test_benchmarks.py::test_solver[benchmark_bci-shallowfbcspnet] **FAILED** [ 96%]

😢

**Comment by tomMoral** [comment]
Hey @bruAristimunha 

Don't worry about this error, I will try to see what is happening here. Not sure why there is no log, this looks like a super weird one with the CI (working on my machine too). I might just skip the test if it is too complicated but I will give it another try.

**Comment by bruAristimunha** [comment]
I supported skip this test and open a future issue to solve this.

**Comment by tomMoral** [comment]
Ok I found the issue.
The issue is that we run out of IO space on the device (see annotations at the bottom of [this page](https://github.com/benchopt/benchmark_bci/actions/runs/8966277140)).

I will try to empty the `__cache__` folder.

**Comment by tomMoral** [comment]
The new issue is due to a warning on OSX which breaks checking the benchopt version in conda envs.
benchopt/benchopt#709 should fix this, testing it here.

File: solvers/ShallowFBCSPNet.py [hunk 2]
```diff
@@ -56,6 +46,8 @@ def set_objective(self, X, y, sfreq):
         y: training labels to train the model.
         sfreq: sampling frequency to allow filtering the data.
         """
+        import faulthandler
+        faulthandler.dump_traceback_later(30, exit=True)
```
Comment (tomMoral) [inline solvers/ShallowFBCSPNet.py:50]:
```suggestion
```

**Comment by tomMoral** [comment]
Ok it should be green now!!! :heavy_check_mark:

**Comment by bruAristimunha** [comment]
Amazing @tomMoral!!! 

Thank you so much for all your work, so nice to pair with you :)

**Pull request merged**